docs/handbooks/state_transition.md

```diff
@@ -0,0 +1,190 @@
+1. **slot processing**, performed for every slot (12 seconds).
+2. **epoch processing**, performed every epoch (32 slots).
+3. **block processing**, performed each time a block is received.
+
+> [!IMPORTANT]  
+> From the Eth2Book:
+> Although the beacon chain's state transition is conceptually slot-driven, as the spec is written a state transition is triggered by receiving a block to process. That means that we first need to fast-forward from our current slot number in the state (which is the slot at which we last processed a block) to the slot of the block we are processing. We treat intervening slots, if any, as empty. This "fast-forward" is done by [`process_slots()`](https://eth2book.info/capella/part3/transition/#def_process_slots), which also triggers epoch processing as required.
+> 
+> In actual client implementations, state updates will usually be time-based, triggered by moving to the next slot if a block has not been received. However, the fast-forward functionality will be used when exploring different forks in the block tree.
+
+The state transition part of a consensus client can be considered as black box, no other parts of the client need to be understood (thoroughly) to be able to make an implementation, furthermore it can be used as a standalone component.
+## Decomposition
+
+The goal through decomposition is to enable the thorough understanding of the inner workings of the function. Going through each argument and line one by one. To not get too much into the weeds we will only go 1 layer deep into the function.
+### Declaration
+
+The `state_transition` function is declared as follows:
+
+```python
+def state_transition(state: BeaconState, signed_block: SignedBeaconBlock, validate_result: bool=True) -> None:
+    block = signed_block.message
+    # Process slots (including those with no blocks) since block
+    process_slots(state, block.slot)
+    # Verify signature
+    if validate_result:
+        assert verify_block_signature(state, signed_block)
+    # Process block
+    process_block(state, block)
+    # Verify state root
+    if validate_result:
+        assert block.state_root == hash_tree_root(state)
+```
+
+### Signature
+
+| Parameter       | Type                 | Description                                 |
+|-----------------|----------------------|---------------------------------------------|
+| `state`         | `BeaconState`        | Describes the current state of the client.  |
+| `signed_block`  | `SignedBeaconBlock`  | Represents the newly received block.        |
+| `validate_result` | `boolean`          | If validation is required; default is true.     |
+| Return Value    | `None`               | As it's a pure function, no value is returned. |
+
+>[!IMPORTANT]
+>In this example the return value is `None` because of the use of Python being an OOP language. Elixir being a functional language makes:
+>1. data immutable
+>2. mandatory for functions to return
```

Comment:
you could return "nil" in Elixir. The reason we change the signature is only because data is immutable and we can't change the content of the beacon state.

Reply:
I see, I have added the info related to pure functions and immutability.